### PR TITLE
Add hd44780-driver to embedded-hal-driver section

### PR DESCRIPTION
**Pull request by JohnDoneth**
https://github.com/JohnDoneth/hd44780-hal

### CODE REVIEW

**Comment by JohnDoneth** [comment]
Right, ouch. My mistake, that totally makes sense, that was a bit of misunderstanding on my part combined with not knowing what to name the crate. Is there a way to rename a crate besides migrating to a new one?

The main difference between this crate and the other (clerk) is that this crate uses only the embedded-hal digital I/O traits to provide functionality while clerk uses it's own traits which have be wrapped around an existing interface such as sysfs_gpio.

**Comment by JohnDoneth** [comment]
I have renamed and moved the crate to [hd44780-hal](https://github.com/JohnDoneth/hd44780-hal) as well as edited this pull request to reflect the change.

File: README.md [hunk 1]
```diff
@@ -303,7 +303,7 @@ Work in progress drivers. Help the authors make these crates awesome!
 - [INA260] - I2C - power monitor - ![crates.io](https://img.shields.io/crates/v/ina260.svg)
 - [ILI9341] - SPI - TFT LCD display
 - [HD44780] - Parallel port - LCD controller
-- [HD44780-hal] - GPIO - LCD controller - ![crates.io](https://img.shields.io/crates/v/hd44780-hal)
+- [HD44780-driver] - GPIO - LCD controller - ![crates.io](https://img.shields.io/crates/v/hd44780-driver)
```
Comment (berkus) [inline README.md]: Shouldn't this link have .svg extension like others do?

File: README.md [hunk 2]
```diff
@@ -340,6 +341,7 @@ Work in progress drivers. Help the authors make these crates awesome!
 [INA260]: https://crates.io/crates/ina260
 [ILI9341]: https://github.com/yuri91/ili9341-rs
 [HD44780]: http://github.com/kunerd/clerk
+[HD44780-driver]: https://github.com/JohnDoneth/hd44780-driver
```
Comment (therealprof) [inline README.md]: Would you please change this link to the crates.io link?

**Pull request merged**